kadi/commands/tests/test_commands.py

```diff
@@ -3,52 +3,99 @@
 
 import numpy as np
 from astropy.table import Table
+import astropy.units as u
 import pytest
 
 # Use data file from parse_cm.test for get_cmds_from_backstop test.
 # This package is a dependency
 import parse_cm.tests
+from testr.test_helper import has_internet
 from Chandra.Time import secs2date
 from cxotime import CxoTime
 
-# Import cmds module directly (not kadi.cmds package, which is from ... import cmds)
-from .. import commands
-from ... import update_cmds
-
+from kadi import commands
+from kadi.commands import (core, commands_v1, commands_v2, conf, get_starcats,
+                           get_observations)
+from kadi.scripts import update_cmds_v1, update_cmds_v2
+from kadi.commands.command_sets import get_cmds_from_event
 
 HAS_MPDIR = Path(os.environ['SKA'], 'data', 'mpcrit1', 'mplogs', '2020').exists()
+HAS_INTERNET = has_internet()
+VERSIONS = ['1', '2'] if HAS_INTERNET else ['1']
+
+
+@pytest.fixture(scope="module", params=VERSIONS)
+def version(request):
+    return request.param
+
+
+@pytest.fixture
+def version_env(monkeypatch, version):
+    if version is None:
+        monkeypatch.delenv('KADI_COMMANDS_VERSION', raising=False)
+    else:
+        monkeypatch.setenv('KADI_COMMANDS_VERSION', version)
+    return version
 
 
-def test_find():
-    cs = commands._find('2012:029:12:00:00', '2012:030:12:00:00')
+@pytest.fixture(scope="module", autouse=True)
+def cmds_dir(tmp_path_factory):
+    with commands_v2.conf.set_temp('cache_loads_in_astropy_cache', True):
+        with commands_v2.conf.set_temp('clean_loads_dir', False):
+            cmds_dir = tmp_path_factory.mktemp('cmds_dir')
+            with commands_v2.conf.set_temp('commands_dir', str(cmds_dir)):
+                yield
+
+
+def test_find(version):
+    if version == '1':
+        idx_cmds = commands_v1.IDX_CMDS
+        pars_dict = commands_v1.PARS_DICT
+    else:
+        idx_cmds = commands_v2.IDX_CMDS
+        pars_dict = commands_v2.PARS_DICT
+
+    cs = core._find('2012:029:12:00:00', '2012:030:12:00:00',
+                    idx_cmds=idx_cmds, pars_dict=pars_dict)
     assert isinstance(cs, Table)
-    assert len(cs) == 147
-    assert np.all(cs['timeline_id'][:10] == 426098447)
-    assert np.all(cs['timeline_id'][-10:] == 426098448)
+    assert len(cs) == 147 if version == '1' else 151  # OBS commands in v2 only
+    if version == '1':
+        assert np.all(cs['timeline_id'][:10] == 426098447)
+        assert np.all(cs['timeline_id'][-10:] == 426098448)
+    else:
+        assert np.all(cs['source'][:10] == 'JAN2612A')
+        assert np.all(cs['source'][-10:] == 'JAN3012C')
     assert cs['date'][0] == '2012:029:13:00:00.000'
     assert cs['date'][-1] == '2012:030:11:00:01.285'
     assert cs['tlmsid'][-1] == 'CTXBON'
 
-    cs = commands._find('2012:029:12:00:00', '2012:030:12:00:00', type='simtrans')
+    cs = core._find('2012:029:12:00:00', '2012:030:12:00:00', type='simtrans',
+                    idx_cmds=idx_cmds, pars_dict=pars_dict)
```

Comment:
Might be safer to version these tests too (if version 1 etc) instead of using the idx_cmds and pars_dict.  Even if _find is not a public method.

Reply:
Sorry, I don't know what you mean, please try again.

Reply:
What I was trying to say is that for many of the tests, the diff was easy to see as the "v1" version basically just copied the old test into an if/else.  Here, the code in the test test is now changed to pass idx_cmds and pars_dict (and what those are and what they do isn't really defined in _find()) so it is harder to see if the "v1" test is testing the "same" thing.

Reply:
OK I see. Because of the split V1 and V2, the new `idx_cmds` and `pars_dict` args are actually required args to `_find`. The values being provided in this test for V1 correspond precisely to what was hardwired in the old version (current release/master) of `_find` in the old `commands.py`. So it is not possible to replicate exactly in code the old test because of this API change, but examination of the old/new code shows easily that the test is doing the same thing.